src/main/java/gregtech/common/terminal/app/configurator/ConfiguratorApp.java

```diff
@@ -0,0 +1,370 @@
+                        log.add("- Missing (" + templateCover.getPickItem().getDisplayName() + ")");
+                        continue;
+                    }
+
+                    // If current machine has no cover or has a different cover
+                    if (!preexistingCover || !coversAreEqual) {
+                        // Create template cover NBT and paste onto existing cover
+                        if (existingTileEntity.placeCoverOnSide(side, templateCoverItemStack, templateCover.getCoverDefinition(), entityPlayer)) {
+                            // If filter is present, remove from player inventory or else remove from cover
+                            if (!SubtractFilterIfMissing(entityPlayer.inventory.mainInventory, templateCover, existingTileEntity.getCoverAtSide(side))) {
+                                log.add("- Missing filter for (" + templateCover.getPickItem().getDisplayName() + ")");
+                            }
+                            if (!RemoveItemFromInventory(entityPlayer.inventory.mainInventory, templateCoverItemStack)) {
+                                log.add("- Missing (" + templateCover.getPickItem().getDisplayName() + ")");
+                            }
+                        } else {
+                            log.add("- Missing (" + templateCover.getPickItem().getDisplayName() + ")");
+                            continue;
+                        }
+                    } else {
+                        if (!SubtractFilterIfMissing(entityPlayer.inventory.mainInventory, templateCover, existingTileEntity.getCoverAtSide(side))) {
+                            log.add("- Missing filter for (" + templateCover.getPickItem().getDisplayName() + ")");
+
+                        }
+                    }
+                }
+
+                // Finally copy settings onto the cover
+                NBTTagCompound templateCoverNBT = new NBTTagCompound();
+                templateCover.writeToNBT(templateCoverNBT);
+                CoverBehavior newCover = existingTileEntity.getCoverAtSide(side);
+                newCover.readFromNBT(templateCoverNBT);
+
+                // Special Cases
+                if (newCover instanceof CoverMachineController) {
+                    // Save control covers for later
+                    ControlCovers.add((CoverMachineController) newCover);
+                } else if (newCover instanceof CoverCraftingTable) {
+                    // Clear machine inventory of crafting table to avoid duping items
+                    ((CoverCraftingTable) newCover).clearMachineInventory(NonNullList.create());
+                }
+            }
+        }
+        // TODO: Redundant?
+        // Ensure the covers being controlled still exist
+        for (CoverMachineController controlCover : ControlCovers) {
+            if (existingTileEntity.getCoverAtSide(controlCover.getControllerMode().side) == null) {
```

Comment:
This should also check if `controlCover.getControllerMode().side != null` before trying to check for the cover at the side. The Machine Controller's side can be null if it is controlling the machine itself, and not any covers on the machine.